src/input/shared.rs

```diff
@@ -114,6 +114,9 @@ fn clean_int_str(mut s: &str) -> Option<Cow<str>> {
     // strip leading and trailing whitespace
     s = s.trim();
 
+    // strip leading unary plus
+    s = s.trim_start_matches('+');
```

Comment:
This will allow `+++0`, for example, which is probably best to reject?

I think probably better to do this like:

```suggestion
    s = s.strip_prefix('+').unwrap_or(s);
```

Worth adding a test case for multiple `+` signs producing an error too.

Reply:
Sounds reasonable :+1:

Reply:
Couldn't get your suggestion into the same commit as a test (or at least I don't know how to do that via the github interface), so I just copied the same thing locally.

However.. Maybe there is a need for some more handling of _weird_ prefix combinations? Not even python likes it when I ask it to parse `+-1`, which this change would technically allow.

Reply:
No nevermind, it seems to fail as expected, I will commit a test for that too.

I cannot take the credit for that though, I really expected it to parse the int.